lib/src/analyzers/lint_analyzer/lint_analysis_config.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:glob/glob.dart';
+
+import 'anti_patterns/models/obsolete_pattern.dart';
+import 'metrics/models/metric.dart';
+import 'rules/models/rule.dart';
+
+class LintAnalysisConfig {
```

Comment:
@incendial please add `@immutable` annotation, add some documentation like:

This object represent merged configs and operates with our business objects.

Reply:
Done.